src/Core/StaticConfigTrait.php

```diff
@@ -112,13 +112,31 @@ public static function setConfig($key, $config = null): void
      * Reads existing configuration.
      *
      * @param string $key The name of the configuration.
-     * @return mixed Configuration data at the named key or null if the key does not exist.
+     * @return mixed|null Configuration data at the named key or null if the key does not exist.
      */
     public static function getConfig(string $key)
     {
         return static::$_config[$key] ?? null;
     }
 
+    /**
+     * Reads existing configuration for a specific key.
+     *
+     * The config value for this key must exist, it can never be null.
+     *
+     * @param string $key The name of the configuration.
+     * @return mixed|null Configuration data at the named key.
```

Comment:
The `|null` should be removed there too.

Reply:
If psalm doesnt fail now, woudnt these help people to use the correct methods as per use case a bit more cleanly? without it they could accidently use the wrong one, with this they have the chance to either use nullable and check for empty, or use the orFail() one with clear exception path.
What do you think?

Reply:
Pslam doesn't report any failure because there are new usages of this method in the core. If you use the return value of this method for a non nullable argument it will complain. Same for `InstanceConfigTrait::getConfigOrFail()`.

Reply:
Exactly, and those new code pieces should be using the correct method already. To avoid someone the config not being filled this way, and silently dragging along this longer than necessary.

Reply:
If others feel also that we "dont care" here, then we can also revert this one, but it sure wouldn't help in terms of strictness and helping to prevent bugs and alike.

Reply:
Huh?? This is the new `getConfigOrFail()` method you added. The description says "The config value for this key must exist, **it can never be null**" and on the other hand the return type claims it **can** be `null`. So the type in docblock contradicts the description, how does that make any sense?

Reply:
Oh sry, did I rebase that wrongly and now the typehint was added to the wrong method?
You are right here, I am sry, I didnt have the diff in front of me for the discussion.